docs/content/guides/developer/app-examples/plinko.mdx

```diff
@@ -0,0 +1,772 @@
+    /// The house's stake is also added to the game's stake.
+    fun internal_start_game(counter: &mut Counter, num_balls: u64, coin: Coin<SUI>, house_data: &HouseData, fee_bp: u16, ctx: &mut TxContext): (ID, Game) {
+        // Ensure guess is valid.
+        // map_guess(guess);
+        let user_stake = coin::value(&coin);
+        // Ensure that the stake is not higher than the max stake.
+        assert!(user_stake <= hd::max_stake(house_data), EStakeTooHigh);
+        // Ensure that the stake is not lower than the min stake.
+        assert!(user_stake >= hd::min_stake(house_data), EStakeTooLow);
+        // Ensure that the house has enough balance to play for this game.
+        assert!(hd::balance(house_data) >= (user_stake*(*vector::borrow(&hd::multiplier(house_data), 0)))/100, EInsufficientHouseBalance);
+
+        let vrf_input = counter_nft::get_vrf_input_and_increment(counter, num_balls);
+
+        let id = object::new(ctx);
+        let game_id = object::uid_to_inner(&id);
+
+        let new_game = Game {
+            id,
+            game_start_epoch: tx_context::epoch(ctx),
+            stake: coin::into_balance<SUI>(coin),
+            player: tx_context::sender(ctx),
+            vrf_input,
+            fee_bp
+        };
+
+        emit(NewGame {
+            game_id,
+            player: tx_context::sender(ctx),
+            vrf_input,
+            user_stake,
+            fee_bp
+        });
+
+        (game_id, new_game)
+    }
+
+}
+```
+
+#### Constants
+
+- `GAME_RETURN`: Used in fee calculations. Might be removed in the current repo.
+
+#### Error codes
+
+Error handling is integral to the module, with specific codes indicating various failure states or invalid operations:
+
+- `EStakeTooLow`: Indicates that the stake provided is below the minimum threshold.
+- `EStakeTooHigh`: Indicates that the stake exceeds the maximum allowed limit.
+- `EInvalidBlsSig`: Denotes an invalid BLS signature.
+- `EInsufficientHouseBalance`: Indicates the house does not have enough balance to cover the game's outcome.
+- `EGameDoesNotExist`: Used when a referenced game cannot be found.
+
+#### Events
+
+- `NewGame`: Emitted when a new game starts, capturing essential details like game ID, player address, VRF input, stake, and fee basis points.
+- `Outcome`: Emitted upon the conclusion of a game, detailing the outcome, including the game ID, result, player address, and a trace of the game's execution.
+
+#### Structures
+
+- `Game`: Represents an individual game session, holding information such as the game ID, epoch of game start, stake amount, player address, VRF input, and the fee basis points.
+
+#### Entry functions
+
+- `start_game`: Initiates a new Plinko game session, accepting parameters like a counter NFT, stake, house data, and transaction context.
+- `finish_game`: Calculates and finalizes the game's outcome, distributing the total funds to the player based on the balls' outcomes.
```

Comment:
- `finish_game`: Calculates and finalizes the game's outcome, the trace path the balls and distributes the total funds to the player based on the balls' outcomes.